runner/scheduler.py

```diff
@@ -481,6 +481,21 @@ def run(self):
         self.progress_dialog.close()
         self.save_runstats()
         self.reset_time()
+
+        # Now create all the reference data using the master log
+        reference_data = []
+        for example in self.tests:
+            reference_data.extend(
+                example.reference_data_from_log(list(self.master_log.test_events(example.unique_id))))
+
+        reference_data_path = RETRIEVAL_REFERENCE_DIR.joinpath(self.config.run_name, "reference_data.json")
+        reference_data_path.parent.mkdir(parents=True, exist_ok=True)
+        with open(reference_data_path, 'w') as f:
+            json.dump(reference_data, f, indent=2)
+
+        if self.agent.retrieval_evaluator:
+            self.agent.retrieval_evaluator.output(self.config.run_name)
+
```

Comment:
If this doesn't take long and can be performed with the data available in the master log, it could be better to just have it extracted somewhere else, probably in a dedicated script? That way we can also spare saving that file and perform the eval on any benchmark log.